File: serialization/src/main/java/net/corda/v5/serialization/SerializedBytes.java
package net.corda.v5.serialization;

import net.corda.v5.base.annotations.CordaSerializable;
import net.corda.v5.base.annotations.DoNotImplement;
import org.jetbrains.annotations.NotNull;

/**
 * A type safe wrapper around a byte array that contains a serialised object.
 */
@SuppressWarnings("unused")
@DoNotImplement
@CordaSerializable
public interface SerializedBytes<T> {

    @NotNull
    byte[] getBytes();
}

File: serialization/src/main/kotlin/net/corda/v5/serialization/SerializedBytes.kt
package net.corda.v5.serialization

import net.corda.v5.base.annotations.CordaSerializable
import net.corda.v5.base.types.OpaqueBytes

/**
 * A type safe wrapper around a byte array that contains a serialised object.
 */
@Suppress("unused")
@CordaSerializable
class SerializedBytes<T : Any>(bytes: ByteArray) : OpaqueBytes(bytes) {
    val summary: String get() = "SerializedBytes(size = $size)"
}
